ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/hateoas/Service.java

```diff
@@ -36,7 +36,9 @@ public void addResource(String resourceName, Resource resource) {
             hateoasResourceMapping.get(resourceName).add(resource);
             return;
         }
-        hateoasResourceMapping.put(resourceName, Arrays.asList(resource));
+        List<Resource> hateoasResources = new ArrayList<>();
+        hateoasResources.add(resource);
+        hateoasResourceMapping.put(resourceName, hateoasResources);
```

Comment:
Wouldn't this be a cleaner implementation for the `addResource` util? 

```java
public void addResource(String resourceName, Resource resource) {
    List<Resource> resources = hateoasResourceMapping.getOrDefault(resourceName, new ArrayList<>());
    resources.add(resource);
    hateoasResourceMapping.put(resourceName, resources);
}
```

Reply:
Fixed with [cc560b5f45cc22da05e188998e11470aad370e83](https://github.com/ballerina-platform/openapi-tools/pull/1643/commits/cc560b5f45cc22da05e188998e11470aad370e83)